### PR TITLE
Change generate curry file path from bin directory to project directory for matching the project structure

### PR DESCRIPTION
**Pull request by zhongwuzw**

### CODE REVIEW

File: bin/generate [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env xcrun swift
 
-// Generates a Swift file with implementation of function currying for a ridicolously high number of arguments
+// Generates a Swift file with implementation of function currying for a ridiculously high number of arguments
```
Comment (mokagio) [inline bin/generate:3]: Thanks for this 😄 .

File: bin/generate [hunk 2]
```diff
@@ -71,11 +71,11 @@ let curries = (start...limit+1).map { curryDefinitionGenerator(arguments: $0) }
 let output = curries.joined(separator: "\n\n") + "\n"
 
 let outputPath = "Source/Curry.swift"
-let currentPath = URL(fileURLWithPath: FileManager.default.currentDirectoryPath)
-let currySwiftPath = currentPath.appendingPathComponent(outputPath)
+let parentPath = URL(fileURLWithPath: FileManager.default.currentDirectoryPath).deletingLastPathComponent()
```
Comment (mokagio) [inline bin/generate]: I'm afraid this approach could backfire. In fact, if I try it on my machine, where the repo is located at

```
~/Developer/open-source/Curry
```

what I get is (_indented for convenience_)

```
~/D/o/Curry (remotes/zhongwuzw/master) ❯❯❯ rm Source/Curry.swift
remove Source/Curry.swift? y

~/D/o/Curry (remotes/zhongwuzw/master ●1) ❯❯❯ git status
HEAD detached at 48fca58
Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	deleted:    Source/Curry.swift

no changes added to commit (use "git add" and/or "git commit -a")

~/D/o/Curry (remotes/zhongwuzw/master) ❯❯❯ bin/generate
Generating 💬
An error occurred while saving the generated functions. Error: 
	Error Domain=NSCocoaErrorDomain
	Code=4 "The folder “Curry.swift” doesn’t exist."
	UserInfo= { 
		NSFilePath=/Users/gio/Developer/open-source/Source/Curry.swift,
		NSUserStringVariant=Folder,
		NSUnderlyingError=0x7faac064ceb0 {
			Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"
		}
	}
```

On the other hand if I try the same combo on `master (386b060)` it works ok.

```
~/D/o/Curry (master) ❯❯❯ rm Source/Curry.swift
remove Source/Curry.swift? y

~/D/o/Curry (master ●1) ❯❯❯ git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	deleted:    Source/Curry.swift

no changes added to commit (use "git add" and/or "git commit -a")

~/D/o/Curry (master ●1) ❯❯❯ bin/generate
Generating 💬
Done, curry functions files written at Source/Curry.swift 👍

~/D/o/Curry (master) ❯❯❯ git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
```

**Review by mokagio — COMMENTED** [review]
@zhongwuzw thanks for this PR 😄 .

I'm not sure I understand the problem you are trying to solve though 🤔 Please refer to my comments.

**Comment by gfontenot** [comment]
Oops, I've been meaning to comment on this for months now. Sorry.

My assumption here is that this changes the expectation to be that you run this script from _inside_ `bin`, which generally isn't the case. If that's the intent behind this change, then I don't think it's what we want, but I'll happily accept a PR to fix the spelling error.

**Comment by zhongwuzw** [comment]
@gfontenot , I updated the code that we can run script from everywhere.

**Comment by gfontenot** [comment]
I went ahead and merged a commit with the spelling fix, but I'm going to close this for now without merging the behavior changes. I'm still unclear on what the problem we're trying to solve is, to be honest. If you want to re-open with these changes to restart discussion I'd be open to it.

**Pull request closed without merging**